src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.model.Book;
+
+public class BookDaoImpl implements BookDao {
+    private static final String URL = "jdbc:mysql://localhost:3306/homework";
+    private static final String USER = "root";
+    private static final String PASSWORD = "vlad77kk";
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?,?)";
+        try (Connection connection = DriverManager.getConnection(URL, USER, PASSWORD);
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows == 0) {
+                throw new RuntimeException("Creating book failed, no ID obtained");
+            }
+            try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
```

Comment:
you don't need the inner `try (ResultSet rs = ...) {` because

[A ResultSet object is automatically closed by the Statement object that generated it](https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#close%28%29)